src/runtime/components/server/destroy_component.cpp

```diff
@@ -4,20 +4,72 @@
 //  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
 #include <hpx/config.hpp>
-#include <hpx/runtime/naming/name.hpp>
+#include <hpx/async.hpp>
+#include <hpx/runtime/actions/plain_action.hpp>
+#include <hpx/runtime/agas/interface.hpp>
 #include <hpx/runtime/components/server/destroy_component.hpp>
-#include <hpx/runtime/components/stubs/runtime_support.hpp>
+#include <hpx/runtime/components/server/memory_block.hpp>
+#include <hpx/runtime/naming/name.hpp>
+
+#include <hpx/util/logging.hpp>
+
+#include <cstdint>
+
+HPX_PLAIN_ACTION_ID(hpx::components::server::destroy_component,
+    hpx_destroy_component_action, hpx::actions::free_component_action_id);
 
 ///////////////////////////////////////////////////////////////////////////////
 namespace hpx { namespace components { namespace server
 {
     ///////////////////////////////////////////////////////////////////////////
     void destroy_component(naming::gid_type const& gid,
-        naming::address const& addr, error_code& ec)
+        naming::address const& addr_)
     {
-        using components::stubs::runtime_support;
-        agas::gva g (addr.locality_, addr.type_, 1, addr.address_);
-        runtime_support::free_component_sync(g, gid, 1);
+        naming::address addr(addr_);
```

Comment:
Why do you need a local copy?

Reply:
The local copy is needed because `agas::is_local_address_cached` requires an lvalue.